Sources/Datadog/Core/System/MobileDevice.swift

```diff
@@ -99,3 +100,29 @@ internal class MobileDevice {
         }
     }
 }
+
+private class LowPowerModeMonitor {
+    
+    var isLowPowerModeEnabled = false
+    private var powerStateDidChangeObserver: Any?
+    
+    init(_ processInfo: ProcessInfo) {
+        isLowPowerModeEnabled = processInfo.isLowPowerModeEnabled
+        powerStateDidChangeObserver = NotificationCenter
+            .default
+            .addObserver(forName: .NSProcessInfoPowerStateDidChange,
+                         object: nil,
+                         queue: .main) { [weak self] notification in
+                guard let processInfo = notification.object as? ProcessInfo else {
+                    return
+                }
+                self?.isLowPowerModeEnabled = processInfo.isLowPowerModeEnabled
```

Comment:
iOS fires this notification (so we write to it) in the main thread whereas we read it in background threads ,`UploadWorker`s have their own `DispatchQueue`s in which they read `MobileDevice.current`.

if read and write happen at the same time the app may crash. can we do read/write in the same `DispatchQueue`?

we have [a class called `ValuePublisher`](https://github.com/DataDog/dd-sdk-ios/blob/master/Sources/Datadog/Core/Utils/ValuePublisher.swift#L22) which can be used for this, but it doesn't exist in your branch.
can you rebase onto `master`? or you'd like to have this fix as a hotfix to `v1.5.2`?

Reply:
I've rebased onto `master` and updated to use a ValuePublisher - thanks for the prompts on the usual practices in this codebase 👍